### PR TITLE
feat: `--eof` flag and config key

### LINKED ISSUE #8997: feat(forge): add --eof flag for eof support
### Component

Forge

### Describe the feature you would like

For alphanet users that want to compile for eof, it would be great to add a flag which configures solc for eof (assuming docker under the hood). 

Previously we asked users to 
- checkout this repo https://github.com/paradigmxyz/forge-alphanet
- such that config available for eof 

Would be nicer for alphanet users to add an additional flag that would compile for eof. 

### Additional context

Couple of things to note: 
- There may be lots of changes with eof, so would be good to make explicit that this is experimental/unstable flag. 
- When users run for the first time, it would download a docker image and might take a while, we can highlight this in the docs, just something to think about

### PR DESCRIPTION
**Pull request by klkvr**
## Motivation

Closes #8997 

## Solution

Adds `--eof` flag to `CoreBuildArgs` and `eof` config key. If set, enables via_ir, sets evm_version to prague and instead of using usual solc binaries, configures solc to pull and run docker image instead.

I've added logic to display a warning on first installation (basically if `solc --version` takes too long), so it should be less confusing when forge just hangs on the first `--eof` run

I've added a note to flag description that this is a temporary approach for EOF support, imo we can keep `--eof` flag name, and just adjust its behavior later, for example only enable `eofVersion` and `viaIr` instead of doing docker stuff

### CODE REVIEW

**Review by grandizzy — APPROVED** [review]
lgtm, tested locally, also with docker service stopped / removed, error messages makes sense

**Review by jenpaff — APPROVED** [review]
lgtm! nice work

File: crates/config/src/lib.rs [hunk 1]
```diff
@@ -904,6 +930,40 @@ impl Config {
...
+            return match rx.recv_timeout(Duration::from_secs(1)) {
+                Ok(res) => res,
+                Err(RecvTimeoutError::Timeout) => {
+                    eprintln!(
+                        "{}",
+                        yansi::Paint::yellow(
+                            "Pulling Docker image for eof-solc, this might take some time..."
+                        )
+                    );
+                    rx.recv().expect("sender dropped")
```
Comment (mattsse) [inline crates/config/src/lib.rs:962]: ah this is clever

**Review by mattsse — APPROVED** [review]

File: crates/forge/tests/cli/alphanet.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Ensure we can run basic counter tests with EOF support.
+#[cfg(not(windows))]
+forgetest_init!(test_eof_flag, |prj, cmd| {
```
Comment (DaniPopes) [inline crates/forge/tests/cli/alphanet.rs:3]: This test fails on macos CI https://github.com/foundry-rs/foundry/actions/runs/11158029813/job/31013451682

**Pull request merged**